### ISSUE TITLE
`omicron-dev run-all` relies on `CRDB_SEED_DIR`

### ISSUE THREAD

**Issue by david-crespo** [oxidecomputer/omicron#4206]
Recent changes to crdb seeding stuff mean that `omicron-dev run-all` relies on an env var that is usually set by nextest. Ideally, `omicron-dev run-all` would handle this somehow and keep working as a one-liner.

```
~/oxide/omicron $ cargo run --bin=omicron-dev -- run-all
    Finished dev [unoptimized + debuginfo] target(s) in 1.04s
     Running `target/debug/omicron-dev run-all`
omicron-dev: setting up all services ...
log file: /var/folders/lk/vsx86g0545g3g2tsmvs50py40000gn/T/omicron-dev-omicron-dev.26093.0.log
note: configured to log to "/var/folders/lk/vsx86g0545g3g2tsmvs50py40000gn/T/omicron-dev-omicron-dev.26093.0.log"
thread 'main' panicked at 'CRDB_SEED_DIR missing -- are you running this test with `cargo nextest run`?: NotPresent', nexus/test-utils/src/db.rs:22:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
zsh: abort      cargo run --bin=omicron-dev -- run-all
```

Indeed, I can make things work by running the seed command manually to get the env var and then manually setting it:

```
~/oxide/omicron $ cargo run -p crdb-seed 
    Finished dev [unoptimized + debuginfo] target(s) in 0.68s
     Running `target/debug/crdb-seed`
Oct 04 21:15:18.620 INFO Using existing seed directory: `/var/folders/lk/vsx86g0545g3g2tsmvs50py40000gn/T/crdb-base/c57f4ced1447ac7b5e0722b644563fcf94b4998591ad01a8c2cabf9ec1af380e`
Oct 04 21:15:18.620 WARN NEXTEST_ENV not set (is this script running under nextest?)

~/oxide/omicron $ CRDB_SEED_DIR=/var/folders/lk/vsx86g0545g3g2tsmvs50py40000gn/T/crdb-base/c57f4ced1447ac7b5e0722b644563fcf94b4998591ad01a8c2cabf9ec1af380e cargo run --bin=omicron-dev -- run-all
    Finished dev [unoptimized + debuginfo] target(s) in 0.92s
     Running `target/debug/omicron-dev run-all`
omicron-dev: setting up all services ...
log file: /var/folders/lk/vsx86g0545g3g2tsmvs50py40000gn/T/omicron-dev-omicron-dev.26123.0.log
note: configured to log to "/var/folders/lk/vsx86g0545g3g2tsmvs50py40000gn/T/omicron-dev-omicron-dev.26123.0.log"
DB URL: postgresql://root@[::1]:58862/omicron?sslmode=disable
DB address: [::1]:58862
omicron-dev: services are running.
omicron-dev: nexus external API:    127.0.0.1:12220
omicron-dev: nexus internal API:    [::1]:12221
etc etc 
```

This is the spot that relies on the env var. I'd try a fix myself, but I'm not sure what the best approach is. I assume manual runs of `run-all` should do something analogous to `cargo run -p crdb-seed` in order to get the seed dir location and pass it to the rest of the DB setup.

https://github.com/oxidecomputer/omicron/blob/a9104a0786f437e38d4945dd77470bc8fd4f974c/nexus/test-utils/src/db.rs#L19-L35

**Comment by davepacheco** [comment]
Ouch.  Also unfortunate here is that there's a test for `omicron-dev run-all` but I bet it succeeds erroneously because the subprocess inherits the CRDB_SEED_DIR environment variable.  We should probably run the subprocess in a clean environment to detect issues like this.

**Comment by sunshowers** [comment]
My bad, haha. We can turn crdb-seed into a library/binary combo pretty easily.

**Comment by sunshowers** [comment]
I'll include this in the fix for #4208.
